pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -142,14 +144,33 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 		err = NewSignupError("multiple matching MasterUserRecord resources found")
 		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, instance, r.setStatusInvalidMURState, err, "Multiple MasterUserRecords found")
 	} else if len(murs) == 1 {
+		mur := murs[0]
+
+		// If the user has been deactivated, then we need to delete the MUR
+		if instance.Spec.Deactivated {
+			err = r.client.Delete(context.TODO(), &mur)
+			if err != nil {
+				return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, instance, r.setStatusFailedToDeleteMUR, err,
+					"Error creating MasterUserRecord")
+			}
+
+			reqLogger.Info("Deleted MasterUserRecord", "Name", mur.Name)
+			return reconcile.Result{}, r.updateStatus(reqLogger, instance, r.setStatusDeactivated)
```

Comment:
The MUR has not been deleted yet actually. So, instead of setting status to deactivated right away after we send the delete request I think our status management should be a bit more complex.
If MUR exists then:
1. Delete MUR
2. Update status to Deactivating, exit reconcile

When the MUR is actually deleted then another reconcile will be triggered and since there is no MUR anymore the status will be updated to Deactivated.

Reply:
That's reasonable, I've simply deleted the status update from here since we will catch it in the subsequent reconcile further down in the code.

Reply:
By why not setting `Deactivating` reason?
Just thinking about the scenario when something goes wrong during deactivation:
T0. User provisioned. Signup CR status set to Complete with ApprovedManually/Automaticaly reason (here is a bug in our code! we set "" as the reason when complete==true)
T1. Set spec.Deactivated = true
T2. Cntrl reconcile requests MUR to be deleted and currently it doesn't update the status so it's still Complete=true and the original reason (which should be Approved when we fix the bug in T0)
T3. Something goes wrong and MUR is not deleted. Then we are stuck with wrong Status from T0.

So, I would do the following:
- Fix Approved reason/status for T0.
- Then in T3: Set Complete to False with reason: Deactiavating.

Reply:
I've made the changes in T3, I think they are a good idea so now Complete is set to false with a reason of "Deactivating".  I don't agree with T0 though, remember that UserSignup has two status conditions, one for `Approved` and one for `Complete`.  The `Approved` status will already contain a reason indicating whether the signup was approved automatically or manually, and it would be redundant to duplicate this information for the `Complete` reason also.

Reply:
OK I forgot that we keep Approved status in a separate condition. Just looks a bit weird that we have an empty reason when Complete is false. But I think that’s ok if we have only one state when Complete == true (it’s when MUR is created).